File: src/main/java/com/refinedmods/refinedstorage/item/property/NetworkItemPropertyGetter.java
package com.refinedmods.refinedstorage.item.property;

import com.refinedmods.refinedstorage.item.NetworkItem;
import net.minecraft.client.multiplayer.ClientLevel;
import net.minecraft.client.renderer.item.ClampedItemPropertyFunction;
import net.minecraft.client.renderer.item.ItemPropertyFunction;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.item.ItemStack;

import javax.annotation.Nullable;

public class NetworkItemPropertyGetter implements ClampedItemPropertyFunction {
    @Override
    public float unclampedCall(ItemStack stack, @Nullable ClientLevel level, @Nullable LivingEntity entity, int p) {
        return entity != null && NetworkItem.isValid(stack) ? 1.0f : 0.0f;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/item/property/NetworkItemPropertyGetter.kt
package com.refinedmods.refinedstorage.item.property

import com.refinedmods.refinedstorage.item.NetworkItem
import net.minecraft.client.world.ClientWorld
import net.minecraft.entity.LivingEntity
import net.minecraft.item.IItemPropertyGetter
import net.minecraft.item.ItemStack

class NetworkItemPropertyGetter : IItemPropertyGetter {
    fun call(stack: ItemStack, @Nullable world: ClientWorld?, @Nullable entity: LivingEntity?): Float {
        return if (entity != null && NetworkItem.Companion.isValid(stack)) 1.0f else 0.0f
    }
}